### ISSUE TITLE
helm-x-path-walker not work

### ISSUE THREAD

**Issue by snowman** [Lompik/x-path-walker#5]
When opening a JSON file, `M-x helm-x-path-walker RET` show list of path, but when `RET` or `C-j` it changes major-mode to `fundamental-mode`, and modify the buffer.

Cursor change only works for some key of JSON file.

Here is an example of a JSON file:
```json
{
  "name": "__MSG_appName__",
  "version": "0.0.1",
  "manifest_version": 2,
  "description": "__MSG_appDescription__",
  "icons": {
    "16": "images/icon-16.png",
    "128": "images/icon-128.png"
  }
}
```

`M-x helm-x-path-walker RET` not walk point for key such as input `icons 16`, `name`, `version`.
but walk point for input `manifest_version`.

In summary, it
1. may or may not change point
2. changes major-mode
3. modify buffer content
